tooling/validateSBOM.sh

```diff
@@ -0,0 +1,246 @@
+    exit 1
+  fi
+
+  if [ -z "$WORKSPACE" ]; then
+    echo "validateSBOM.sh: WARNING: WORKSPACE environment variable not detected."
+    echo "validateSBOM.sh: Using ${ORIGIN}/sbom_validation"
+    WORKSPACE_DIR="${ORIGIN}/sbom_validation"
+  else
+    WORKSPACE_DIR="${WORKSPACE}/sbom_validation"
+  fi
+
+  if [ -d "${WORKSPACE_DIR}" ]; then
+    echo "validateSBOM.sh: New temporary workspace already exists in ${WORKSPACE_DIR}"
+    echo "Refreshing directory to avoid conflict."
+    rm -rf "${WORKSPACE_DIR}"
+  fi
+  echo "validateSBOM.sh: Setting up workspace directory ${WORKSPACE_DIR}"
+  mkdir "${WORKSPACE_DIR}"
+
+  cd "${WORKSPACE_DIR}" || exit 1
+
+  JDK_MAJOR_VERSION="$1"
+  SOURCE_TAG="$2"
+  SBOM_LOCATION="$3"
+
+  if ! echo "$JDK_MAJOR_VERSION" | grep -q "^[1-9][0-9]*\$"; then
+    echo "ERROR: validateSBOM.sh: first argument must be a positive integer greater than 0."
+    exit 1
+  fi
+
+  if [ -z "$SOURCE_TAG" ]; then
+    echo "ERROR: validateSBOM.sh: second argument must not be empty."
+    exit 1
+  fi
+  
+  TAG_CHECK=""
+  if [ "$JDK_MAJOR_VERSION" -eq "8" ]; then
+    echo "$SOURCE_TAG" | grep -q "^jdk8u[0-9][0-9]*-b[0-9][0-9]*_adopt\$"
+    TAG_CHECK="$?"
+    echo "$SOURCE_TAG" | grep -q "^jdk8u[0-9][0-9]*-b[0-9][0-9]*\$"
+    TAG_CHECK="${TAG_CHECK}$?"
+    echo "$SOURCE_TAG" | grep -q "^jdk8u[0-9][0-9]*-ga\$"
+    TAG_CHECK="${TAG_CHECK}$?"
+    echo "$SOURCE_TAG" | grep -q "^jdk8u[0-9][0-9]*-dryrun-ga\$"
+    TAG_CHECK="${TAG_CHECK}$?"
+    echo "$SOURCE_TAG" | grep -q "^jdk8u[0-9][0-9]*-aarch32-[0-9][0-9]*\$"
+    TAG_CHECK="${TAG_CHECK}$?"
+    echo "$SOURCE_TAG" | grep -q "^jdk8u[0-9][0-9]*-ga-aarch32-[0-9][0-9]*\$"
+    TAG_CHECK="${TAG_CHECK}$?"
+    echo "$SOURCE_TAG" | grep -q "^jdk8u[0-9][0-9]*-dryrun-ga-aarch32-[0-9][0-9]*\$"
+    TAG_CHECK="${TAG_CHECK}$?"
+  else
+    echo "$SOURCE_TAG" | grep -q "^jdk-[0-9][0-9\.\+]*_adopt\$"
+    TAG_CHECK="$?"
+    echo "$SOURCE_TAG" | grep -q "^jdk-[0-9][0-9\.\+]*\$"
+    TAG_CHECK="${TAG_CHECK}$?"
+    echo "$SOURCE_TAG" | grep -q "^jdk-[0-9][0-9\.\+]*-dryrun-ga\$"
+    TAG_CHECK="${TAG_CHECK}$?"
+    echo "$SOURCE_TAG" | grep -q "^jdk-[0-9][0-9\.\+]*-ga\$"
+    TAG_CHECK="${TAG_CHECK}$?"
+  fi
+  
+  if ! echo "${TAG_CHECK}" | grep -q "0"; then
+    echo "WARNING: validateSBOM.sh: SOURCE_TAG does not use a valid upstream tag structure."
+    echo "INFO: validateSBOM.sh: Build is presumed to be a personal or dev build."
+    echo "INFO: validateSBOM.sh: SCM and SHA checks will be skipped."
+    SOURCE_TAG="null"
+  fi
+
+  if [ -z "$SBOM_LOCATION" ]; then
+    echo "ERROR: validateSBOM.sh: third argument must not be empty."
+    exit 1
+  fi
+
+  # Now we check that the third argument is a valid link.
+  if echo "$SBOM_LOCATION" | grep -q "^https.*"; then
+    if ! curl -L "$SBOM_LOCATION" -O; then
+      echo "ERROR: SBOM_LOCATION could not be downloaded." 
+      exit 1
+    fi
+    SBOM_NAME=$(echo "${SBOM_LOCATION}" | sed 's:.*/::')
```

Comment:
This is slightly more efficient as it doesn't spawn other processes, but I'm ok wither way.
```suggestion
    SBOM_NAME=$(SBOM_LOCATION ##*/}
```

Reply:
I had to tweak this a little, but I see what you meant. Change made.